File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int max = nums.length;
        int[] res = new int[2];
        for(int num : nums){
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        for(int i = 1; i <= max; ++i){
            if(!map.containsKey(i)) res[1] = i;
            else if(map.get(i) == 2) res[0] = i;
        }
        
        return res;
        
    }
}

File: 0645-set-mismatch/0645-set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n_map = [0] * (len(nums) + 1)
        
        for i in range(len(nums)):
            n_map[nums[i]] += 1
        
        zero, one = 0, 0
        for i in range(1, len(n_map)):
            if n_map[i] == 0:
                one = i
            
            if n_map[i] > 1:
                zero = i
        
        return [zero, one]